src/foundry/client/applications/ux/search-filter.d.mts

```diff
@@ -1,9 +1,142 @@
+import type { ValueOf } from "#utils";
+
 /**
  * A controller class for managing a text input widget that filters the contents of some other UI element.
- * @remarks TODO: Stub, copy from v12 implementation & update
  */
-declare class SearchFilter {}
+declare class SearchFilter {
+  /**
+   * @param options - Options which customize the behavior of the filter
+   */
+  constructor(options: SearchFilter.Configuration);
+
+  /**
+   * The value of the current query string
+   */
+  query: string;
+
+  /**
+   * A callback function to trigger when the tab is changed
+   */
+  callback: SearchFilter.Callback;
+
+  /**
+   * The allowed Filter Operators which can be used to define a search filter
+   */
+  static OPERATORS: Readonly<{
+    EQUALS: "equals";
+    CONTAINS: "contains";
+    STARTS_WITH: "starts_with";
+    ENDS_WITH: "ends_with";
+    LESS_THAN: "lt";
+    LESS_THAN_EQUAL: "lte";
+    GREATER_THAN: "gt";
+    GREATER_THAN_EQUAL: "gte";
+    BETWEEN: "between";
+    IS_EMPTY: "is_empty";
+  }>;
+
+  /**
+   * The regular expression corresponding to the query that should be matched against
+   * @defaultValue `undefined`
+   */
+  rgx: RegExp | undefined;
+
+  /**
+   * A reference to the HTML navigation element the tab controller is bound to
+   */
+  _input: HTMLElement | null;
+
+  /**
+   * Bind the SearchFilter controller to an HTML application
+   */
+  bind(html: HTMLElement): void;
+
+  /**
+   * Release all bound HTML elements and reset the query.
+   */
+  unbind(): void;
+
+  /**
+   * Perform a filtering of the content by invoking the callback function
+   * @param event - The triggering keyboard event
+   * @param query - The input search string
+   */
+  filter(event: KeyboardEvent, query: string): void;
+
+  /**
+   * Clean a query term to standardize it for matching.
+   * See https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/normalize
+   * @param query - An input string which may contain leading/trailing spaces or diacritics
+   * @returns A cleaned string of ASCII characters for comparison
+   */
+  static cleanQuery(query: string): string;
+
+  /**
+   * A helper method to test a value against a precomposed regex pattern.
+   * @param rgx   - The regular expression to test
+   * @param value - THe value to test against
+   * @returns Does the query match?
+   */
+  static testQuery(rgx: RegExp, value: string): boolean;
+
+  /**
+   * Test whether a given object matches a provided filter
+   * @param obj    - An object to test against
+   * @param filter - The filter to test
+   * @returns Whether the object matches the filter
+   */
+  static evaluateFilter(obj: Record<string, any>, filter: SearchFilter.FieldFilter): boolean;
```

Comment:
Why `Record<string, any>`?

`AnyObject` works for plain objects, `Record<string, any>` allows objects, classes, instances, and functions.